### ISSUE TITLE
Cannot remove questions when combining questions to create multi-series charts

### ISSUE THREAD

**Issue by jeff-bruemmer** [metabase/metabase#12794]
**Describe the bug**

When adding additional questions to an existing question to create a multi-series chart, you cannot remove a question by clicking on the `X` next to the question title above the chart. The sidebar checkboxes, however, function as expected, i.e. users can add and remove questions by checking and unchecking sidebar boxes.

**Logs**
What happens when clicking the `X` next to a question:

```
app-main.bundle.js?a9635a44a9161a01fd75:5 Error getting setting graph.dimensions TypeError: Cannot read property 'filter' of null
    at N (app-main.bundle.js?a9635a44a9161a01fd75:5)
    at he (app-main.bundle.js?a9635a44a9161a01fd75:5)
    at Object.getDefault (app-main.bundle.js?a9635a44a9161a01fd75:5)
    at g (app-main.bundle.js?a9635a44a9161a01fd75:5)
    at g (app-main.bundle.js?a9635a44a9161a01fd75:5)
    at t.getComputedSettings (app-main.bundle.js?a9635a44a9161a01fd75:5)
    at h (app-main.bundle.js?a9635a44a9161a01fd75:5)
    at Function.value (app-main.bundle.js?a9635a44a9161a01fd75:5)
    at app-main.bundle.js?a9635a44a9161a01fd75:5
    at Array.filter (<anonymous>)

And many more cannot read ... of null errors.

```
**To Reproduce**

When editing a question on a dashboard, you can add additional questions by clicking on the add icon.

If you select a question from the sidebar by checking its box, Metabase adds the question to the chart as expected.

However, if you try to remove the question by clicking on the `X` next to the question above the chart, nothing happens (except errors logged to the console).

Unchecking the box next to the question in the sidebar does correctly remove the question, so users at least still have a way to add and remove questions.

**Expected behavior**
Clicking on the `X` next to the question should remove the question.

**Screenshots**

The inert `X`es are next to the currently selected questions above the chart:

![cannot_delete_question_by_clicking_on_x](https://user-images.githubusercontent.com/17343328/85884636-c74edb00-b797-11ea-81e4-ef1e3662c39f.png)

**Severity**
Small bug. Just unexpected UI behavior, and users can still remove questions via the checkboxes in the sidebar.

The sluggishness of the sidebar is a bigger usability issue.

**Additional context**

**Metabase Diagnostic Info**

```json
{
  "browser-info": {
    "language": "en-US",
    "platform": "MacIntel",
    "userAgent": "Mozilla/5.0 (Macintosh; Intel Mac OS X 10_15_5) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/83.0.4103.116 Safari/537.36",
    "vendor": "Google Inc."
  },
  "system-info": {
    "file.encoding": "UTF-8",
    "java.runtime.name": "OpenJDK Runtime Environment",
    "java.runtime.version": "11.0.7+10",
    "java.vendor": "AdoptOpenJDK",
    "java.vendor.url": "https://adoptopenjdk.net/",
    "java.version": "11.0.7",
    "java.vm.name": "OpenJDK 64-Bit Server VM",
    "java.vm.version": "11.0.7+10",
    "os.name": "Linux",
    "os.version": "4.14.181-108.257.amzn1.x86_64",
    "user.language": "en",
    "user.timezone": "GMT"
  },
  "metabase-info": {
    "databases": [
      "googleanalytics",
      "mysql",
      "druid",
      "postgres",
      "bigquery",
      "redshift",
      "mongo",
      "h2"
    ],
    "hosting-env": "unknown",
    "application-database": "postgres",
    "application-database-details": {
      "database": {
        "name": "PostgreSQL",
        "version": "9.6.11"
      },
      "jdbc-driver": {
        "name": "PostgreSQL JDBC Driver",
        "version": "42.2.8"
      }
    },
    "run-mode": "prod",
    "version": {
      "date": "2020-06-26",
      "tag": "v0.36.0-snapshot",
      "branch": "master",
      "hash": "bc9e3db"
    },
    "settings": {
      "report-timezone": "US/Pacific"
    }
  }
}
```

**Comment by nitsujri** [comment]
@flamber thanks for linking these issues!

One difference between this issue and https://github.com/metabase/metabase/issues/17394 and https://github.com/metabase/metabase/issues/21149 is that this issue is removing the whole `question` vs removing a specific series/column from a `question`.

**Comment by flamber** [comment]
@nitsujri Then that would be a feature request, which would first require this issue to be fixed - #2697.

**Comment by MartinCura** [comment]
Having the same problem on v0.42.3

![image](https://user-images.githubusercontent.com/14017665/182461641-0ddfc8d5-028d-4d15-b823-5a9a474c1edf.png)

For me this is pretty annoying because we have several SQL Questions with field filters, so i was trying to avoid having many copies by exposing all series from the same Question but hiding them in certain graphs (not sure if this will be quicker, whether it reuses the same query between instances of the same Question in a Dashboard, but at least is more maintainable).

But as i can't hide any series, as i tried here, this is not possible.

Any help?
